WordPressAuthenticator/OAuth/OAuthTokenRequestBody.swift

```diff
@@ -0,0 +1,46 @@
+/// Models the request to send for an OAuth token
+///
+/// - Note: See documentation at https://developers.google.com/identity/protocols/oauth2/native-app#exchange-authorization-code
+struct OAuthTokenRequestBody: Encodable {
+    let clientId: String
+    let clientSecret: String
+    let code: String
+    let codeVerifier: String
+    let grantType: String
+    let redirectURI: String
+
+    enum CodingKeys: String, CodingKey {
+        case clientId = "client_id"
+        case clientSecret = "client_secret"
+        case code
+        case codeVerifier = "code_verifier"
+        case grantType = "grant_type"
+        case redirectURI = "redirect_uri"
+    }
+
+    func asURLEncodedData() throws -> Data {
+        let params = [
+            (CodingKeys.clientId.rawValue, clientId),
+            (CodingKeys.clientSecret.rawValue, clientSecret),
+            (CodingKeys.code.rawValue, code),
+            (CodingKeys.codeVerifier.rawValue, codeVerifier),
+            (CodingKeys.grantType.rawValue, grantType),
+            (CodingKeys.redirectURI.rawValue, redirectURI),
+        ]
+
+        let items = params.map { URLQueryItem(name: $0.0, value: $0.1) }
+
+        var components = URLComponents()
+        components.queryItems = items
+
+        guard let query = components.query else {
```

Comment:
I don't think it's possible to have a `nil` query in this case?

Reply:
Yes. That should be the case.

🤔 I guess I'm just uncomfortable force unwrapping stuff, but it does make for simpler code in a case like this one.

I tried to write a test that resulted in this code throwing, but I wasn't able to. That's not to say that there is _no way_ for the failure to occur, of course. Still, I feel a bit more confident about removing the errors here.

Reply:
Addressed in 8b6f6a89dd39665b903b73871e6f5c327f30770b